### DISCUSSION TITLE
Is VVC supposed to look softer (i.e. encode less efficiently) than x.265 for low resolution videos at relatively low bitrates?

### DISCUSSION THREAD

**Discussion by birdie-github** [fraunhoferhhi/vvenc#389] [Q&A]
That's being discussed [here](https://forum.doom9.org/showthread.php?p=2003146#post2003146).

All the files are found at the link.

I'm a little bit confused that at low resolutions (720p)/relatively low bitrates (4759 kb/s) vvenc loses to x265.

**Comment by lehmann-c** [comment]
1. Your source is already compressed with h264 - so it´s already compressed with artifacts and is blurry. Any codec applied to this can not get any better.  
2. The png that you created out of this 1 second video clip is just some frame (frame 23/30).  
If you want to compare single frames, you have to either encode just single frames or configure the codec to use Intra-Frames only. 
In your comparison you just picked a picture in between the GOP-structure. That means depending on what picture type the encoder have chosen for that particular frame it can be better or worse quality than an different encoder (or even the same codec with a different configuration, like intra-only or low-latency mode).   
Video codec quality assessment is always applied on sequences not single frames.  You have to encode single frames (or intra only) if you want to compare single frames.  
In your example, frame 23 is hierarchically in temporal layer 5. That means, it is a b-frame with many dependencies to other frames at a very low bitrate.

**Reply by birdie-github** [reply]
1. While this is true, for almost any codec given enough bitrate, its output can become practically lossless for all intents and purposes. In my example the source doesn't seem to have any H.264 encoding artifacts or maybe I'm just not seeing them. Would be great if you showed that this clip is not "lossless" if you have enough time and will.

Actually I've done re-encoding of H.264 on multiple occasions while shaving off 50-75% of bitrate while largely preserving the source video quality. HW H.264 encoders for instance do not use B frames and that's enough to get major bitrate savings.

2. That's absolutely true but you make it sound like VVC (or vvenc) excels at I frames while its B frames could be much worse. x265 was in the exact same situation for this clip and we are talking about the same type of frame.

> Video codec quality assessment is always applied on sequences not single frames.

Aren't new more advanced codecs supposed to make each frame look better at the same bitrate than older codecs? That's clearly not the case for this clip where each x265 frame looks sharper than vvenc output.

> In your example, frame 23 is hierarchically in temporal layer 5. That means, it is a b-frame with many dependencies to other frames at a very low bitrate.

Isn't it exactly the same situation for x265?

Again, I'm not trying to prove x265 is better than vvenc at relatively low resolutions and bitrates, I'm trying to find the ground truth and get better results from vvenc. I don't have a RED camera and terabytes of storage to shoot in visually lossless formats.

**Reply by birdie-github** [reply]
Here's another thing:

VVC could be eventually used for video conferencing. Most webcams produce either MJPEG or H.264 output. That doesn't bode well for the codec which loses to HEVC in this scenario.

**Comment by birdie-github** [comment]
Even here https://videocodecs.info/vivict/?clipname=dnd&leftVideoUrl=%2Fdata%2Fdnd%2Fdnd_vvenc-h266_slow%2Fdnd.vvenc-h266_slow_g128_q24.h266.llproxy.webm&rightVideoUrl=%2Fdata%2Fdnd%2Fdnd_libx265_veryslow%2Fdnd.libx265_veryslow_g128_crf26.mp4.llproxy.webm

x265 veryslow beats vvenc slow at three or even four top quality settings. There's just more details to the former.

Check the face circa 4 seconds into the clip. Check the beard and the skin texture. VVENC blurs everything out.

It's only at extremely bitrate starved quality settings that vvenc starts to look better, that's

vvenc slow 34 571.9kbps vs x265 veryslow 36 685.7kbps
